src/octonode/client.coffee

```diff
@@ -142,37 +144,106 @@ class Client
         body = JSON.parse(body || '{}')
       catch err
         return callback(err)
-
+    debugger
     return callback(new HttpError(body.message, res.statusCode, res.headers, body)) if body.message and res.statusCode in [400, 401, 403, 404, 410, 422]
     callback null, res.statusCode, body, res.headers
 
+  errorHandlePromise: (response, cb) ->
+    debugger
+    {statusCode, body, headers} = response
+
+    fiveError = new HttpError('Error ' + statusCode, statusCode, headers)
+    fourError = new HttpError(body.message, statusCode, headers, body)
+
+    return cb(fiveError) if Math.floor(statusCode/100) is 5
+    return cb(fourError) if body.message and Math.floor(statusCode/100) is 4
+
+    return cb(null, statusCode, body, headers)
+
   # Github api GET request
-  get: (path, params..., callback) ->
-    @request @requestOptions(
+
+  get: (path, params..., cb) =>
+
+    cb = if cb then cb else () -> arguments
+
+    options = @requestOptions(
       uri: @buildUrl path, params...
       method: 'GET'
-    ), (err, res, body) =>
-      return callback(err) if err
-      @errorHandle res, body, callback
-
-  # Github api GET request
-  getNoFollow: (path, params..., callback) ->
-    @request @requestOptions(
+      followRedirect: true
+      json: true
+      simple: false
+      resolveWithFullResponse: true
+    )
+
+    return @rp options
+      .then((response) => @errorHandlePromise(response, cb))
+      .catch((err) -> cb(err))
```

Comment:
I think I need to re-work the error handling. Right now the successful response is coupled with http errors. I'm going to use the default request-promise functionality that rejects the promise on http errors (moving to catch) and only resolves on a 2xx response. I'll have an errorHandle and a successHandle function that both take the callback as an input and return a function per above.